### ISSUE TITLE
Proposal: get rid of warning on unused parameters

### ISSUE THREAD

**Issue by diggit** [flit/argon-rtos#5]
Hi,
`-Wunused` and for me `-Wunused-parameter` is sometimes very handy.

As title says, it would be nice to have such warnings suppressed on functions, that intentionally not use some parameter.

Possible solutions are explained [here on SO](https://stackoverflow.com/questions/3599160/how-to-suppress-unused-parameter-warnings-in-c/12891181).

If you want this solved, which solution do you prefer?

**Comment by flit** [comment]
Yeah, that's a good idea. I normally have that warning turned off, but I'd be happy to address it.

My preference is the unused attribute via a macro, like the attribute macros at the top of ar_internal.h:

```c
#if !defined(UNUSED)
#define UNUSED __attribute__((unused))
#endif
```

This, of course, can be used in a function definition:
```c
void some_function(int x, UNUSED int y)
{
    return x;
}
```

And it provides an opportunity to customize or set to empty based on the compiler.

I'm not a fan of the `(void)x;` style, as long as the unused attribute is available.

Thanks!

**Comment by diggit** [comment]
Ok, I can fix that, but where to put this macro?
One unused parameter is `width` in classes.h:602 which does not include ar_internal.h (of course, it's public include)
Another case is `object2` in ar_mutex.cpp:15 which `ar_internal.h` includes...

GCC and armcc support `__attribute__((unused))`, not sure about IAR.

**Comment by flit** [comment]
The unused `width` parameter on the `Channel` ctor looks like a bug, actually. It will even cause `TypedChannel` to not work correctly for any type that is not the same size as `void *`. Thanks for finding this! 😀 

So, probably the best place for an UNUSED macro is at the top of `ar_internal.h`, next to the other attribute macros.
